.travis.yml

```diff
@@ -2,7 +2,7 @@ sudo: required
 dist: trusty
 language: cpp
 
-# Install Bazel.
+# Install Bazel and cppcheck.
```

Comment:
sgtm, I'll update to use `clang-tidy` instead.

Reply:
It seems the latest cppcheck is only [just over a month old](http://cppcheck.sourceforge.net/). Given that, I'll probably just add `clang-tidy` as an additional static analysis tool.

Reply:
After looking into this further it seems getting `clang-tidy` to work with Bazel isn't as straightforward as I would like since it requires a compile commands file. Consequently, I'll defer setting up `clang-tidy` for now since we already have `cppcheck`.

Reply:
I think there is a way. I'll look into it soon.

Reply:
Try this out for local testing: https://github.com/grailbio/bazel-compilation-database. clang-tidy has been hands-down the best tool I have used. Don't give up on it just yet!

If you ever do get it working, I recommend adding all the warnings and winnowing them down if you get too many spurious errors. The way to generate the complete config is `clang-tidy -dump-config -checks=* > .clang-tidy`.